crates/hir_ty/src/method_resolution.rs

```diff
@@ -385,7 +388,24 @@ fn iterate_method_candidates_impl(
                     krate,
                     traits_in_scope,
                     name,
-                    callback,
+                    &mut |ty, item, is_inherent| {
```

Comment:
I think it would be nicer to do just pass the module on further and do the check in `is_valid_candidate`?